app/domain/authentication/authn_jwt/fetch_jwt_signing_key_factory.rb

```diff
@@ -0,0 +1,44 @@
+module Authentication
+  module AuthnJwt
+    # Factory that returns the interface implementation of FetchSigningKey
+    class FetchJwtSigningKeyFactory
+
+      def initialize(authenticator_parameters)
+        @authenticator_input = authenticator_parameters
+        @fetch_provider_uri_signing_key = Authentication::AuthnJwt::FetchProviderUriSigningKey.new(@authenticator_input)
+        @fetch_jwks_uri_signing_key = Authentication::AuthnJwt::FetchJwksUriSigningKey.new(@authenticator_input)
+      end
+
+      def create
+        validate_key_configuration
+
+        if provider_uri_is_valid?
+          @fetch_provider_uri_signing_key
+        elsif jwks_uri_is_valid?
+          @fetch_jwks_uri_signing_key
+        end
+      end
```

Comment:
my comment was deleted so i'll write it again - factory classes should not have an initializer and instead have a static method named `create` that should receive the input needed to create a class.

Reply:
sorry for the ping pong between @nessiLahav 's comments and mine

Reply:
regarding these:
```
        @fetch_provider_uri_signing_key = Authentication::AuthnJwt::FetchProviderUriSigningKey.new(@authenticator_input)
        @fetch_jwks_uri_signing_key = Authentication::AuthnJwt::FetchJwksUriSigningKey.new(@authenticator_input)
```

they can be memoized below in methods instead of the initializer

Reply:
So basically ill need to move `authenticator_parameters` to `create` and get rid of `initialize`?

Reply:
correct.